docs/proposals/resource-quota-scoping.md

```diff
@@ -0,0 +1,264 @@
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Resource Quota - Scoping resources
+
+## Problem Description
+
+### Ability to quota compute requests and limits
+
+The existing `ResourceQuota` document constrains the total amount of compute
+resource requests.  This is useful when a `cluster-admin` is interested in
+controlling explicit resource guarantees often associated with higher SLA.
+The end-user creating the pod is expected to have intimate knowledge on 
+their minimum required resource as well as their potential limits.
+
+There are many environments where a `cluster-admin` does not extend this level
+of trust to their end-user because user's often request too much resource, and
+they have trouble reasoning between what they hope to have available for their
+application versus what their application actually needs.  In these environments,
+cluster-managers often just expose a single value (the limit) to the end-user,
+and internally, they set the request as a factor of the limit to control for 
+over-commit or based on known-historical trends.  In this environment, the
+`cluster-admin` is interested in exposing a quota to their end-users that maps
+to their desired limit instead of their request.
+
+### Ability to quota to control footprint
+
+The current `ResourceQuota` document does not allow the ability
+to quota best-effort pods separately from pods with resource
+guarantees.  The `cluster-admin` wants to control footprint
+to fairly share the nodes, but not restrict the ability to
+run best-effort pods.  By definition, a best-effort pod has
+zero footprint.  As a result, the `cluster-admin` requires
+the ability to control the max number of active best-effort pods.
+
+### Ability to quota long-running vs bounded-duration compute resources
+
+The `cluster-admin` may want to quota end-users separately
+based on long-running vs bounded-duration compute resources.
+
+For example, a `cluster-admin` may offer more compute resources
+for long running pods that are expected to have a more permanent footprint
+on the node than bounded-duration pods.
+
+A classic example is a PaaS deployment where the `cluster-admin` may 
+allow a separate budget for pods that run their web application vs pods that
+build web applications.
+
+Another example is providing more quota to a database pod than a
+pod that performs a database migration.
+
```

Comment:
For this whole section, is the key reason admins treat these pods differently in policies due to the duration or is duration merely correlated with importance?  That is, admins set aside fewer guaranteed resources for these short-duration jobs because they can generally afford brief downtime, not merely because they'll finish quickly.  It just so happens that most such jobs can finish quickly.